apps/server/src/modules/task-card/uc/task-card.uc.ts

```diff
@@ -167,9 +143,8 @@ export class TaskCardUc {
 		if (params.dueDate) {
 			card.dueDate = params.dueDate;
 		}
-
-		if (!card.isVisibleBeforeDueDate()) {
-			throw new ValidationError('Invalid date combination');
+		if (course) {
+			card.course = course;
```

Comment:
in the `update` there are if cases for `title`, `duedate` and `course` (`params.courseId` is checked above for existence but is actually also marked as required in the ticket)
even though all of the three are required, so this should be changed.
according to the ticket (and the `ITaskCardCRUD`) only `text` and `visibleAtDate` are optional. The later one also will be changed, but that is not part of this ticket